src/scriptdev2/scripts/eastern_kingdoms/molten_core/boss_ragnaros.cpp

```diff
@@ -169,6 +167,58 @@ struct boss_ragnarosAI : public Scripted_NoMovementAI
             m_uiEnterCombatTimer = 10000;
     }
 
+    // Custom threat management for targets in melee range
+    bool CanMeleeTargetInRange()
```

Comment:
Ok, I looked closer at Unit::SelectAttackingTarget and it should indeed do the trick. I'll test that soon and update the PR accordingly, thanks. 👍